src/libstd/io/mod.rs

```diff
@@ -371,6 +371,14 @@ where
     loop {
         if g.len == g.buf.len() {
             unsafe {
+                // FIXME(danielhenrymantilla): #42788
+                //
+                //   - This creates a (mut) reference to a slice of
+                //     _uninitialized integers_.
+                //
```

Comment:
It's still UB; but in the spec, not rustc, which rustc + standard library (and nobody else) is allowed to take advantage of.

Reply:
That's what I attempted to express with _unstable_, but I guess it's not really accurate since a nightly user can use an `unstable` feature whereas here only `::std` is allowed to rely on it.

Do you want me to rephrase it?

Reply:
Yes please. It should say something like "This is undefined behavior in Rust the language but since we are the standard library, we alone get to take advantage of what rustc actually does".

I think the PR description should also refrain from labeling things as likely / unlikely so that it is clearer to everyone reading the PR who may rely on this that they should stop doing so.